imap_processing/lo/l1b/lo_l1b.py

```diff
@@ -158,6 +163,54 @@ def get_avg_spin_durations(
     return avg_spin_durations
 
 
+def get_spin_angle(l1a_de: xr.Dataset) -> Union[np.ndarray[np.float64], Any]:
+    """
+    Get the spin angle (0 - 360 degrees) for each DE.
+
+    Parameters
+    ----------
+    l1a_de : xarray.Dataset
+        The L1A DE dataset.
+
+    Returns
+    -------
+    spin_angle : np.ndarray
+        The spin angle for each DE.
+    """
+    de_times = l1a_de["de_time"].values
+    # DE Time is 12 bit DN. The max possible value is 4096
+    spin_angle = np.array(de_times / 4096 * 360, dtype=np.float64)
+    return spin_angle
```

Comment:
As an FYI, I think we use `.values` too much everywhere in our codebase, we should try to stick with xarray as much as possible. The return value shouldn't be `Any` here either, it is always an array currently? I think it would then be `xr.DataArray` if you take the below suggestion.
```suggestion
    # DE Time is 12 bit DN. The max possible value is 4096
    spin_angle = l1a_de["de_time"].astype(float) / 4096 * 360
    return spin_angle
```

Reply:
I did that so I wasn't carrying over the coordinate and dimension information that's stored in the de_times data array because I don't want that as part of the spin_angle array. Is there a preferred way of sticking with xarray and not carrying that over into the spin_angle variable?

Reply:
Oh, I see... I'm not sure. This seems to indicate the opposite would happen and they were getting dropped when they wanted them preserved 🤷 
https://stackoverflow.com/questions/70289775/dataarray-deletes-attributes-in-simple-computation

Fine to keep this as-is, my approval still stands.